### PR TITLE
WIP: Drivers

### PR DESCRIPTION
**Pull request by sagikazarmark**

### CODE REVIEW

**Comment by acrobat** [comment]
Alright looks good! So the general "rules" are

- Use the `Bernard\Driver\*DriverName*` namespace
- The class is named `Driver.php` instead of the driver name now
- The class should be final

Is this correct? If so I can convert some other drivers!

**Comment by sagikazarmark** [comment]
Yeah, that sounds about right. We can probably do the name changes in this repo and migrate drivers at the same time.

**Comment by reinfi** [comment]
What is the reason that the new driver class is final?

**Comment by sagikazarmark** [comment]
There is no reason to extend drivers. If there is, we can always open them later.

File: doc/drivers.rst [hunk 1]
```diff
@@ -106,17 +106,17 @@ as appropriate for your use case.
     <?php
     // doctrine.php
 
-    use Bernard\Command\Doctrine as BernardCommands;
+    use Bernard\Driver\Docrtrine\Command as BernardCommands;
```
Comment (bashilbers) [inline doc/drivers.rst]: Typo here? :trollface:

**Comment by bashilbers** [comment]
Might be a little late, but are you going to update the changelog.md file. It's hard for me to keep track on the changes made from alpha8 to alpha9

**Comment by acrobat** [comment]
@bashilbers There is a todo task to write a migration guide (https://github.com/bernardphp/bernard/projects/1#card-7413690).

File: doc/drivers.rst (hunk 1 above)
Comment (sagikazarmark) [reply doc/drivers.rst]: Yup, thanks

**Comment by sagikazarmark** [comment]
There are no significant changes from alpha8 to alpha9, one PR was merged. I tagged the last release to mark it as a final point before breaking everything.

**Comment by bendavies** [comment]
the bundle needs updating to take into account this change, FYI.

**Comment by acrobat** [comment]
yes indeed, but once @sagikazarmark did most of the changes I will dedicate some time in making the bundle compatible with the 1.0 stable release of the lib.

**Comment by sagikazarmark** [comment]
In the meantime the bundle could be locked to the latest alpha to avoid breaks

**Comment by acrobat** [comment]
Yes I will open a PR for that and I think for the stable release we just should release a new major of the bundle. So bundle v2 == pre lib 1.0 support and bundle v3 == lib 1.* support

**Pull request merged**